lib/src/supertokens.dart

```diff
@@ -1,36 +1,61 @@
-import 'dart:async';
+import 'dart:convert';
 
+import 'package:supertokens/src/errors.dart';
+import 'package:supertokens/src/front-token.dart';
 import 'package:supertokens/src/id-refresh-token.dart';
 import 'package:supertokens/src/utilities.dart';
+import 'package:http/http.dart' as http;
+import 'package:supertokens/supertokens.dart';
+
+enum Eventype {
+  SIGN_OUT,
+  REFRESH_SESSION,
+  SESSION_CREATED,
+  ACCESS_TOKEN_PAYLOAD_UPDATED,
+  UNAUTHORISED
+}
+
+enum APIAction { SIGN_OUT, REFRESH_TOKEN }
 
 /// Primary class for the supertokens package
 /// Use [SuperTokens.initialise] to initialise the package, do this before making any network calls
 class SuperTokens {
   static int sessionExpiryStatusCode = 401;
   static bool isInitCalled = false;
-  static String refreshTokenEndpoint = "";
-  static String? apiDomain;
-  static Map<String, String>? refreshAPICustomHeaders;
-
-  /// Initialises the SuperTokens SDK
-  /// Uses the [refreshTokenEndpoint] to make a call to refresh the session when needed,
-  /// the [sessionExpiryStatusCode] is used to determine unauthorised access API errors.
-  /// [refreshAPICustomHeaders] are always sent when calling the refresh endpoint.
-  /// Throws a [FormatException] if an invalid URL is provided for [refreshTokenEndpoint].
-  static void initialise({
-    required String refreshTokenEndpoint,
-    int sessionExpiryStatusCode = 401,
-    Map<String, String> refreshAPICustomHeaders = const {},
+  static String refreshTokenUrl = "";
+  static String signOutUrl = "";
+  static String rid = '';
+  static NormalisedInputType? config;
+
+  static void init({
+    required String apiDomain,
+    String? apiBasePath,
+    int sessionExpiredStatusCode = 401,
+    String? cookieDomain,
+    String? userDefaultdSuiteName,
+    Function(Eventype)? eventHandler,
+    Function(APIAction, http.Request)? preAPIHook,
+    Function(APIAction, http.Request, http.Response)? postAPIHook,
   }) {
```

Comment:
In the example app main dart file call this init function to make sure that the user can initialise SuperTokens without having to pass anything other than apiDomain